services/brig/test/integration/API/User/Account.hs

```diff
@@ -1047,6 +1067,30 @@ testDomainsBlockedForRegistration opts brig = withDomainsBlockedForRegistration
   where
     p email = RequestBodyLBS . encode $ SendActivationCode (Left email) Nothing False
 
+-- | FUTUREWORK: This is not very exhaustive.  You can change the flag in
+-- brig.integration.yaml manually and re-run the integration tests for more coverage.  If you
+-- want to have that coverage permanently, grep this module for @setRestrictUserCreation@ and
+-- change all those tests to run with both settings.  Use 'testRestrictedUserCreation' here as
+-- an instructive example.
+testRestrictedUserCreation :: Opt.Opts -> Brig -> Http ()
+testRestrictedUserCreation opts brig = do
+  let opts' = opts {Opt.optSettings = (Opt.optSettings opts) {Opt.setRestrictUserCreation = Just True}}
+  withSettingsOverrides opts' $ do
+    e <- randomEmail
+    let emailReq = RequestBodyLBS . encode $ object ["email" .= fromEmail e]
+    post (brig . path "/activate/send" . contentJson . body emailReq)
+      !!! (const 200 === statusCode)
+    getActivationCode brig (Left e) >>= \case
+      Nothing -> liftIO $ assertFailure "missing activation key/code"
+      Just (_, c) -> do
+        let Object reg =
+              object
+                [ "name" .= Name "Alice",
+                  "email" .= fromEmail e,
+                  "email_code" .= c
+                ]
+        postUserRegister' reg brig !!! const 403 === statusCode
```

Comment:
We should also assert on the error label here to ensure any future restrictions don't make this test not test the thing we want to test.

Reply:
0091efc62fcfe1b45af21941d0946675dc2dc1ab

Thanks for making me do that!  :)  I am emberrassed to admit I was just lazy.